@xen-orchestra/lite/src/views/new-vm/NewVmView.vue

```diff
@@ -0,0 +1,968 @@
+  installMode: vmState.installMode,
+  installRepository: vmState.selectedVdi as XenApiVdi['$ref'],
+  name_label: vmState.name,
+  template: vmState.new_vm_template?.$ref,
+  vdis: vmState.vdis,
+  vifs: vmState.networkInterfaces.map(net => ({
+    network: net.interface,
+    mac: net.macAddress,
+  })),
+  cpus: vmState.vCPU,
+  vcpusMax: vmState.VCPUs_max,
+  name_description: vmState.description,
+  memory: vmState.ram,
+  autoPoweron: vmState.auto_power,
+  bootAfterCreate: vmState.boot_vm,
+  copyHostBiosStrings: vmState.boot_firmware !== 'uefi' && !templateHasBiosStrings.value && vmState.copyHostBiosStrings,
+  hvmBootFirmware: vmState.boot_firmware,
+  tags: vmState.tags,
+  cloudConfig: '',
+}))
+
+const xapi = useXenApiStore().getXapi()
+
+const createVM = async () => {
+  const templateRef = vmCreationParams.value.template
+  const newVmName = vmCreationParams.value.name_label
+  const selectedVdiRef = vmCreationParams.value.installRepository
+  const newVdis = vmCreationParams.value.vdis
+  const existingDisks = vmCreationParams.value.existingDisks
+
+  const $defer = []
+
+  if (!templateRef) {
+    console.error('Error : templateRef is undefined or invalid.')
+    return
+  }
+
+  isBusy.value = true
+
+  try {
+    const vmRefs = vmCreationParams.value.clone
+      ? await xapi.vm.clone({ [templateRef]: newVmName })
+      : await xapi.vm.copy({ [templateRef]: newVmName }, '' as XenApiSr['$ref'])
+
+    $defer.push(() => xapi.vm.delete(vmRefs))
+
+    // Removes disks from the provision XML, we will create them by ourselves.
+    await xapi.vm.removeFromOtherConfig(vmRefs, 'disks')
+
+    // Inspects the disk configuration contained within the VM's other_config,
+    // creates VDIs and VBDs and then executes any applicable post-install script.
+    await xapi.vm.provision(vmRefs)
+
+    // INSTALL SETTINGS
+    const vm = await xapi.getField<XenApiVm>('VM', vmRefs[0], 'record')
+
+    const installMethod = vmState.selectedVdi ? 'cd' : 'network'
+
+    // TODO maybe improve this part
+    if (vm.domain_type === 'hvm') {
+      if ((newVdis.length === 0 && existingDisks.length === 0) || installMethod === 'network') {
+        const { order } = vm.HVM_boot_params
+        await xapi.call('VM.set_HVM_boot_params', [vmRefs[0], { order: order ? 'n' + order.replace('n', '') : 'ncd' }])
+      }
+    } else {
+      if (vm.PV_bootloader === 'eliloader') {
+        if (installMethod === 'network') {
+          await xapi.call('VM.set_other_config', [vmRefs[0], 'install-repository', undefined])
+        } else if (installMethod === 'cd') {
+          await xapi.call('VM.set_other_config', [vmRefs[0], 'install-repository', 'cdrom'])
+        }
+      }
+    }
+
+    if (installMethod === 'cd') {
+      const vbds = await xapi.getField<XenApiVbd['$ref'][]>('VM', vmRefs[0], 'VBDs')
+
+      for (const vbdRef of vbds) {
+        const type = await xapi.getField<string>('VBD', vbdRef, 'type')
+        if (type === 'CD') {
```

Comment:
Use the enum `VBD_TYPE` from `@vates/types/common`
```suggestion
        if (type === VBD_TYPE.CD) {
```